lib/libbackscrub.h

```diff
@@ -2,53 +2,38 @@
  * Authors - @see AUTHORS file.
 ==============================================================================*/
 
-// tested against tensorflow lite v2.4.1 (static library)
 #ifndef _LIBBACKSCRUB_H
 #define _LIBBACKSCRUB_H
 
 // for cv::Mat and related types
+#include <stdarg.h>
 #include <opencv2/core/core.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
-#include <stdarg.h>
 
-#define TFLITE_MINIMAL_CHECK(x)                              \
-  if (!(x)) {                                                \
-	fprintf(stderr, "Error at %s:%d\n", __FILE__, __LINE__); \
-	exit(1);                                                 \
-  }
+// Return a new (opaque) mask generation context
+extern void *bs_maskgen_new(
+	// Required parameters
+	const char *modelname,
+	size_t threads,
+	size_t width,
+	size_t height,
+	// Optional (nullable) callbacks with caller-provided context
+	// ..debug output
+	void (*ondebug)(void *ctx, const char *fmt, va_list ap),
```

Comment:
Variable argument lists make for bad FFI compat. Accepted for now.

Reply:
Good point, would ellipsis be any better?

Reply:
Actually worse as not every language supports them. Best is to have a fixed interface with just plain data types, structs or pointers. va_list is somewhat dependent on the ABI of the compiler, as is the ellipsis for varargs. But as said, let's leave this for now and see what features we actually need in the debug calls later on.

Reply:
Agreed, formatting within lib.

Reply:
Leaving this for reference as/when we want to port off of Linux:
https://stackoverflow.com/questions/40159892/using-asprintf-on-windows